# Exploración de los datos

## 1. Importar las librerías

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn

## 2. Cargar datos

Cargar los datos del csv

In [2]:
datos_pacientes = pd.read_csv('./data/Datos Lab 1.csv', )

Trabajar sobre una copia de los datos

In [3]:
datos = datos_pacientes.copy()

## 3. Perfilamiento

Se usa la función head() para tomar una muestra de los primero datos y obtener una visión inicial.

In [4]:
datos.head()

,Patient ID,Date of Service,Sex,Age,Weight (kg),Height (m),BMI,Abdominal Circumference (cm),Blood Pressure (mmHg),Total Cholesterol (mg/dL),...,Physical Activity Level,Family History of CVD,Height (cm),Waist-to-Height Ratio,Systolic BP,Diastolic BP,Blood Pressure Category,Estimated LDL (mg/dL),CVD Risk Score,CVD Risk Level
0,isDx5313,"November 08, 2023",M,44.0,114.300,1.720,38.600,100.000,112/83,228.0,...,High,N,172.000,0.581,112.0,83.0,Hypertension Stage 1,121.0,19.880,HIGH
1,LHCK2961,20/03/2024,F,57.0,92.923,1.842,33.116,106.315,101/91,158.0,...,High,Y,184.172,0.577,101.0,91.0,Hypertension Stage 2,57.0,16.833,INTERMEDIARY
2,WjVn1699,2021-05-27,F,NaN,73.400,1.650,27.000,78.100,90/74,135.0,...,High,N,165.000,0.473,90.0,74.0,Normal,45.0,12.600,LOW
3,dCDO1109,"April 18, 2022",F,35.0,113.300,1.780,35.800,79.600,92/89,158.0,...,Moderate,Y,178.000,0.447,92.0,89.0,Hypertension Stage 1,94.0,14.920,HIGH
4,pnpE1080,01/11/2024,F,48.0,102.200,1.750,33.400,106.700,121/68,207.0,...,Low,Y,175.000,0.610,121.0,68.0,Elevated,128.0,18.870,HIGH


También se pueden observar datos aleatorios del dataset para asegurarse que otras posiciones cumplen con el formato. 

In [9]:
display(datos.sample(10))

,Patient ID,Date of Service,Sex,Age,Weight (kg),Height (m),BMI,Abdominal Circumference (cm),Blood Pressure (mmHg),Total Cholesterol (mg/dL),...,Physical Activity Level,Family History of CVD,Height (cm),Waist-to-Height Ratio,Systolic BP,Diastolic BP,Blood Pressure Category,Estimated LDL (mg/dL),CVD Risk Score,CVD Risk Level
1267,rETH6271,24 Dec 24,M,39.0,64.100,1.700,22.200,99.100,113/69,175.0,...,Moderate,Y,170.000,0.583,113.0,69.0,Normal,90.0,13.590,INTERMEDIARY
1373,KmWR1254,28/03/2024,F,57.0,118.900,1.890,33.300,76.700,147/74,193.0,...,Moderate,N,189.000,0.406,147.0,74.0,Hypertension Stage 2,120.0,17.870,INTERMEDIARY
813,lTUG4850,2020-09-09,M,50.0,110.317,1.852,33.401,100.033,110/90,NaN,...,High,Y,185.168,0.540,110.0,90.0,Hypertension Stage 2,158.0,19.680,INTERMEDIARY
874,TObL7108,01-18-2021,M,56.0,55.400,1.870,15.800,99.100,124/78,209.0,...,High,N,187.000,0.530,124.0,78.0,Elevated,141.0,13.540,INTERMEDIARY
306,Qbvz4402,11 Jul 20,F,55.0,115.300,NaN,32.300,85.600,113/98,192.0,...,Low,Y,NaN,0.453,113.0,98.0,Hypertension Stage 2,107.0,17.950,HIGH
1405,vYLW1606,2020-09-23,M,68.0,NaN,1.851,20.474,97.393,160/99,279.0,...,High,Y,185.117,0.526,160.0,99.0,Hypertension Stage 2,NaN,17.675,INTERMEDIARY
146,lyjE0672,"January 13, 2022",F,42.0,104.100,1.880,29.500,83.400,132/86,196.0,...,Low,Y,188.000,NaN,132.0,86.0,Hypertension Stage 1,98.0,16.420,HIGH
576,toIX9000,"November 11, 2022",M,45.0,91.300,1.870,26.100,80.000,102/61,271.0,...,High,N,187.000,0.428,102.0,61.0,Normal,210.0,17.740,HIGH
1374,TDFh4321,07-30-2021,F,69.0,108.428,1.579,30.689,75.798,122/112,159.0,...,Moderate,N,157.940,0.480,122.0,112.0,Hypertension Stage 2,40.0,17.418,HIGH
964,rOMu6070,06-19-2023,M,71.0,86.264,1.694,29.423,119.233,170/62,204.0,...,High,Y,NaN,0.704,170.0,NaN,Hypertension Stage 2,109.0,18.465,INTERMEDIARY


Usando datos.shape se puede conocer información general sobre la cantidad de registros y variables:

In [10]:
datos.shape

(1639, 24)

Existen 1639 registros. Para cada uno de esos registros hay 24 variables que se deben tener en cuenta. 

Para saber información específica sobre el tipo de dato de cada una de las variables se puede hacer lo siguiente:

In [11]:
datos.info()

<class 'pandas.DataFrame'>
RangeIndex: 1639 entries, 0 to 1638
Data columns (total 24 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Patient ID                    1639 non-null   str    
 1   Date of Service               1639 non-null   str    
 2   Sex                           1639 non-null   str    
 3   Age                           1571 non-null   float64
 4   Weight (kg)                   1566 non-null   float64
 5   Height (m)                    1578 non-null   float64
 6   BMI                           1586 non-null   float64
 7   Abdominal Circumference (cm)  1578 non-null   float64
 8   Blood Pressure (mmHg)         1639 non-null   str    
 9   Total Cholesterol (mg/dL)     1571 non-null   float64
 10  HDL (mg/dL)                   1557 non-null   float64
 11  Fasting Blood Sugar (mg/dL)   1585 non-null   float64
 12  Smoking Status                1639 non-null   str    
 13  Diabetes Statu

Ahora sabemos que la mayoría de los datos son de tipo float. Aún así, hay bastantes datos que son de tipo string. Adicionalmente, podemos ver que la mayoría de variables contienen registros nulos. La mayoría no son nulos, igualmente es una cantidad que se debe tener en cuenta. 

Para conocer más sobre el tipo de información se va a acceder al diccionario de los datos:

In [13]:
diccionario_datos = pd.read_excel('./data/DiccPacientes.xlsx')
pd.set_option('display.max_colwidth', None)
diccionario_datos

,Nombre Columna,Tipo de dato,Comentarios
0,Patient ID,String,Identificador del paciente
1,Date of Service,Date,Fecha de la atención
2,Sex,String,"Sexo (Femenino, Masculino)"
3,Age,Integer,Edad
4,Weight (kg),Float,Peso
5,Height (m),Float,Altura
6,BMI,Float,Índice de masa corporal
7,Abdominal Circumference (cm),Float,Circunferencia abdominal
8,Blood Pressure (mmHg),String,"Presión sanguínea, de la forma ""<Presión arterial sistólica>/<Presión arterial diastólica>"""
9,Total Cholesterol (mg/dL),Float,Colesterol total


Combinando esto con la información que encontramos con datos.info(), podemos tener más contexto sobre cada variable diferente y entender por qué cada variable es el tipo que es. Además, acá se puede ver que hay algunas diferencias con el tipo de dato que sale en el diccionario vs el tipo que sale cuando se usa la función info(). 

En específico, unas variables que dicen tipo string según info(), dice que son booleanas o date según el diccionario. En el caso de las que dicen float en info(), algunas son integer en el diccionario.

Para entender aún mejor, se va a usar la función describe():

In [14]:
datos.describe()

,Age,Weight (kg),Height (m),BMI,Abdominal Circumference (cm),Total Cholesterol (mg/dL),HDL (mg/dL),Fasting Blood Sugar (mg/dL),Height (cm),Waist-to-Height Ratio,Systolic BP,Diastolic BP,Estimated LDL (mg/dL),CVD Risk Score
count,1571.000000,1566.000000,1578.000000,1586.000000,1578.000000,1571.000000,1557.000000,1585.000000,1571.000000,1563.000000,1578.000000,1554.000000,1582.000000,1610.000000
mean,46.803186,85.666006,1.757439,28.424744,91.538861,199.043673,56.183558,117.836860,175.770082,0.522440,125.632637,82.887536,113.235896,18.227281
std,13.039479,21.712504,0.118012,7.309275,13.427985,59.388670,16.721702,32.379634,11.695880,0.085692,22.577463,15.503625,61.435291,10.767666
min,6.134000,13.261000,1.371000,4.317000,49.542000,-1.256000,0.008000,15.306000,136.498000,0.250000,49.914000,31.720000,-92.055000,-20.057000
25%,37.000000,67.100000,1.666500,22.600000,79.700000,150.000000,42.000000,92.000000,167.000000,0.453000,108.000000,71.000000,62.000000,15.150000
50%,46.000000,86.314000,1.760000,28.000000,91.200000,199.000000,56.000000,115.000000,176.000000,0.519000,125.000000,82.000000,112.000000,16.967000
75%,55.000000,104.801500,1.850000,33.963000,102.267250,250.000000,70.000000,139.000000,185.000000,0.582000,141.000000,93.000000,159.000000,18.900000
max,89.420000,158.523000,2.146000,53.028000,136.336000,385.679000,110.315000,219.667000,214.394000,0.804000,202.711000,134.066000,317.314000,114.980000


Obervando los datos de esta manera permite generar conclusiones más específicas sobre las tendencias y otros elementos notables dentro de los datos. Estas son algunas de las conclusiones:
- La mayoría de las personas en los datos tienden a ser adultos alrededor de los 45. Aún así, el rango va de 6 a 81.
- Se puede ver que los pesos de las personas tienden a ser bastante alto. Esto puede indicar posibles hábitos de nutrición dentro de la población estudiada. 
- El BMI y el nivel de colesterol tiene sentido si se analiza junto a los pesos, que tienden a ser un poco altos.
- Algunas variables tienen valores extraños. Por ejemplo, el colesterol total tiene al menos un valor negativo, lo cual es imposible. Lo mismo sucede con HDL y riesgo cardiovascular.
- Existe altura tanto en metros como en centímetros. Esto es redundante y en algunos casos los datos se contradicen ligeramente.

En general, los datos tiened a mostrar una población donde existe un alto riesgo cardiovascular por sobrepeso, reflejado por las tendencias altas en las variables que indican esto. 